lib/handlers/injector-sor.ts

```diff
@@ -269,7 +270,15 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
           sourceOfTruthPoolProvider: noCacheV3PoolProvider,
         })
 
-        const v2PoolProvider = new V2PoolProvider(chainId, multicall2Provider)
+        const nonCachingv2PoolProvider = new V2PoolProvider(chainId, multicall2Provider)
+        const cachingV2PoolProvider = new CachingV2PoolProvider(
+            chainId,
+            new V2PoolProvider(chainId, multicall2Provider),
```

Comment:
Can we reference the no caching object here?

Reply:
aw true.